### PR TITLE
[#165632270] Pick QRcode from gallery

### PR DESCRIPTION
**Pull request by Edigio**
Pick and scan QRcode from an image stored on the device

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
## Affected stories

  * 🌟 [#165632270](https://www.pivotaltracker.com/story/show/165632270): Come CIT vorrei potere utilizzare il Qr code non solo come immagine da fotocamera ma anche come immagine da galleria


New dependencies added: <a href='https://www.npmjs.com/package/react-native-image-picker'><code>react-native-image-picker</code></a> and <a href='https://www.npmjs.com/package/react-native-lewin-qrcode'><code>react-native-lewin-qrcode</code></a>.


<h2><a href="https://github.com/react-community/react-native-image-picker">react-native-image-picker</a></h2>
<p>Author: Marc Shilling</p>
<p>Description: A React Native module that allows you to use native UI to select media from the device library or directly from the camera</p>
<p>Homepage: <a href="https://github.com/react-community/react-native-image-picker">https://github.com/react-community/react-native-image-picker</a></p>

<table>
  <thead><tr><th></th><th width="100%"></th></tr></thead>
  <tr><td>Created</td><td>almost 4 years ago</td></tr><tr><td>Last Updated</td><td>about 2 months ago</td></tr><tr><td>License</td><td>MIT</td></tr><tr><td>Maintainers</td><td>6</td></tr><tr><td>Releases</td><td>116</td></tr><tr><td>Keywords</td><td>react-native, react-native-image-picker, react, native, image and picker</td></tr>
</table>

<details>
<summary><code>README</code></summary></br>

# React Native Image Picker

[npm version](https://badge.fury.io/js/react-native-image-picker)
[npm](https://npmcharts.com/compare/react-native-image-picker?minimal=true)
MIT
Platform - Android and iOS
[Gitter chat](https://gitter.im/react-native-image-picker/Lobby)

A React Native module that allows you to use native UI to select a photo/video from the device library or directly from the camera, like so:

| iOS                                                                                                                   | Android                                                                                                                       |
| --------------------------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------- |
|  |  |

#### _Before you open an issue_

This library started as a basic bridge of the native iOS image picker, and I want to keep it that way. As such, functionality beyond what the native `UIImagePickerController` supports will not be supported here. **Multiple image selection, more control over the crop tool, and landscape support** are things missing from the native iOS functionality - **not issues with my library**. If you need these things, [react-native-image-crop-picker](https://github.com/ivpusic/react-native-image-crop-picker) might be a better choice for you.

## Getting Started

```
yarn add react-native-image-picker
react-native link react-native-image-picker
```

You will also need to add `UsageDescription` on iOS and some permissions on Android, refer to the [Install doc](docs/Install.md).

## Usage

```javascript
import ImagePicker from 'react-native-image-picker';

// More info on all the options is below in the API Reference... just some common use cases shown here
const options = {
  title: 'Select Avatar',
  customButtons: [{ name: 'fb', title: 'Choose Photo from Facebook' }],
  storageOptions: {
    skipBackup: true,
    path: 'images',
  },
};

/**
 * The first arg is the options object for customization (it can also be null or omitted for default options),
 * The second arg is the callback which sends object: response (more info in the API Reference)
 */
ImagePicker.showImagePicker(options, (response) => {
  console.log('Response = ', response);

  if (response.didCancel) {
    console.log('User cancelled image picker');
  } else if (response.error) {
    console.log('ImagePicker Error: ', response.error);
  } else if (response.customButton) {
    console.log('User tapped custom button: ', response.customButton);
  } else {
    const source = { uri: response.uri };

    // You can also display the image using data:
    // const source = { uri: 'data:image/jpeg;base64,' + response.data };

    this.setState({
      avatarSource: source,
    });
  }
});
```

Then later, if you want to display this image in your render() method:

```javascript
<Image source={this.state.avatarSource} style={styles.uploadAvatar} />
```

### Directly Launching the Camera or Image Library

To Launch the Camera or Image Library directly (skipping the alert dialog) you can
do the following:

```javascript
// Launch Camera:
ImagePicker.launchCamera(options, (response) => {
  // Same code as in above section!
});

// Open Image Library:
ImagePicker.launchImageLibrary(options, (response) => {
  // Same code as in above section!
});
```

#### Notes

On iOS, don't assume that the absolute uri returned will persist. See [#107](/../../issues/107)

For more, read the [API Reference](docs/Reference.md).

## License

[MIT](LICENSE.md)


</details>




<h2><a href="https://github.com/LewinJun/react-native-lewin-qrcode#readme">react-native-lewin-qrcode</a></h2>
<p>Author: lewin</p>
<p>Description: 二维码识别，选择相册里面的一张照片进行识别二维码</p>
<p>Homepage: <a href="https://github.com/LewinJun/react-native-lewin-qrcode#readme">https://github.com/LewinJun/react-native-lewin-qrcode#readme</a></p>

<table>
  <thead><tr><th></th><th width="100%"></th></tr></thead>
  <tr><td>Created</td><td>about 1 year ago</td></tr><tr><td>Last Updated</td><td>6 months ago</td></tr><tr><td>License</td><td>ISC</td></tr><tr><td>Maintainers</td><td>1</td></tr><tr><td>Releases</td><td>3</td></tr><tr><td>Keywords</td><td>qrcode, react, 二维码, 识别 and native</td></tr>
</table>

<details>
<summary><code>README</code></summary></br>

# react-native-lewin-qrcode
react-native 识别照片二维码

## Table of contents
- [Install](#install)
- [Usage](#usage)

## Install
### 1: yarn add 或者npm install,现在最新版本是1.0.0
`yarn add react-native-lewin-qrcode  `
### 2: yarn install 或 npm install
### 3: react-native link react-native-lewin-qrcode

## Usage
### NOTE: 可以参考Example的App.js中的openPhoto方法

```javascript
import {readerQR} from 'react-native-lewin-qrcode'
//path 图片文件的路径
readerQR(path).then((data)=>{
  Alert.alert('识别结果',data);
}).catch((err)=>{
   Alert.alert('识别失败');
});
```


</details>


<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by Edigio** [comment]
This is the result
qrcode_image

**Comment by Edigio** [comment]
Fixed user interface 
Simulator Screen Shot - iPhone X - 2019-05-31 at 16 18 51

**Comment by Edigio** [comment]
@matteodesanti this is the complete flow

pick_qr_code

**Comment by matteodesanti** [comment]
Great @Edigio ok for me

File: android/app/src/main/java/it/teamdigitale/app/italiaapp/MainApplication.java [hunk 1]
```diff
@@ -44,6 +46,8 @@ public boolean getUseDeveloperSupport() {
     protected List<ReactPackage> getPackages() {
       return Arrays.<ReactPackage>asList(
         new MainReactPackage(),
+            new QRScanReaderPackage(),
```
Comment (fpersico) [inline android/app/src/main/java/it/teamdigitale/app/italiaapp/MainApplication.java]: Fix indentation please.

Comment (Edigio) [reply android/app/src/main/java/it/teamdigitale/app/italiaapp/MainApplication.java]: Fixed

**Pull request merged**